### PR TITLE
Allow __all__ to be typed

### PR DESCRIPTION
**Pull request by sigmavirus24**
Closes gh-1018

### CODE REVIEW

File: pycodestyle.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def lru_cache(maxsize=128):  # noqa as it's a fake implementation.
         'while',
     )))
 )
-DUNDER_REGEX = re.compile(r'^__([^\s]+)__ = ')
+DUNDER_REGEX = re.compile(r"^__([^\s]+)__(:\W+[a-zA-Z.0-9_\[\]\"]+)? = ")
```
Comment (asottile) [inline pycodestyle.py:172]: should the `\W+` here be `\s*` ?

Comment (sigmavirus24) [reply pycodestyle.py:172]: Probably, I was trying to fit this in in 15m after having just woken up so the caffeine hadn't hit yet

Comment (sigmavirus24) [reply pycodestyle.py]: Updated and made it a non-capturing regex as well

**Review by asottile — APPROVED** [review]

**Pull request merged**